### ISSUE TITLE
How to do pointwise in Ranking problem

### ISSUE THREAD

**Issue by yydai** [dmlc/xgboost#5190]
My training parameters like this:

`params_pointwise = {
    "objective": "multi:softmax",
    "eval_metric": "ndcg@10",
    'booster': 'gbtree',
    'eta': 0.27,
    'gamma': 1,
    'min_child_weight': 6,
    'max_depth': 8,
    'subsample': 0.8,
    'num_class': 3
}`

But xgboost  throws the error like:
`b'[10:45:55] /workspace/src/metric/rank_metric.cc:159: Check failed: preds.size() == info.labels_.size() (68958831 vs. 22986277) label size predict size not match`

My dataset like:
ID       label
id1     2
id1     1
id2     0
id2     2
id2     1

**Comment by trivialfis** [comment]
I think the metric is not prepared for using with multi-class objective.  Output prediction for multi-class is a vector of shape: `num_classes * num_rows`.  In short, I don't think point-wise is supported right now unless you roll out a customized metric.

**Comment by trivialfis** [comment]
Out of curiosity, is pointwise used in practice?  Some interesting ideas in ordinal regression area might be useful for ranking, but I don't see anyone directly cast ranking into reg/cls in practice.

**Comment by yydai** [comment]
Thanks for you answer. Pointwise is not used in production. I am just curious how to do pointwise with XGBoost and I think regression is enough for this job.
